File: internal/models/scram_user.go
package models

import "time"

// ScramUser SCRAM 用户模型
type ScramUser struct {
	UserID     int64      `gorm:"primaryKey;autoIncrement" json:"user_id"`
	ClusterID  int64      `gorm:"not null;index" json:"cluster_id"`
	Username   string     `gorm:"size:256;not null" json:"username"`
	Mechanism  string     `gorm:"size:32;not null;default:'SCRAM-SHA-256'" json:"mechanism"` // SCRAM-SHA-256 或 SCRAM-SHA-512
	SyncStatus string     `gorm:"size:32;default:'synced'" json:"sync_status"`
	LastSyncAt *time.Time `json:"last_sync_at"`
	CreatedAt  time.Time  `json:"created_at"`
	UpdatedAt  time.Time  `json:"updated_at"`
}

// TableName 指定表名
func (ScramUser) TableName() string {
	return "scram_users"
}

File: frontend/src/services/scramUser.ts
import request from './api'

export interface ScramUser {
  user_id: number
  cluster_id: number
  username: string
  mechanism: string
  sync_status: string
  last_sync_at: string | null
  created_at: string
  updated_at: string
}

export interface CreateScramUserRequest {
  cluster_id: number
  username: string
  password: string
  mechanism?: string
}

export const scramUserService = {
  list: async (clusterId: number, page: number = 1, pageSize: number = 20) => {
    const response = await request.get('/scram-users', {
      params: { cluster_id: clusterId, offset: (page - 1) * pageSize, limit: pageSize },
    })
    return response.data
  },

  create: async (data: CreateScramUserRequest) => {
    const response = await request.post('/scram-users', data)
    return response.data
  },

  delete: async (username: string, clusterId: number) => {
    const response = await request.delete(`/scram-users/${encodeURIComponent(username)}`, {
      params: { cluster_id: clusterId },
    })
    return response.data
  },

  sync: async (clusterId: number) => {
    const response = await request.post(`/scram-users/sync/${clusterId}`)
    return response.data
  },
}
